`RollingUpdateTest.shouldPerformRollingUpdate` flaky

**Summary**

> Try to answer the following as best as possible

- How often does the test fail? at least once https://ci.zeebe.camunda.cloud/job/camunda/job/zeebe/job/main/566/testReport/junit/io.camunda.zeebe.test/RollingUpdateTest(it-testrun)/Verify___IT___Test___shouldPerformRollingUpdate__Flaky_Test_/
- Does it block your work? yes
- Do we suspect that it is a real failure? unknown

happened before:
#5003, #5035 (didn't want to reopen the issue as it is 2 years old already)

**Failures**

<details><summary>Example assertion failure</summary>
<pre>
Caused by: java.lang.AssertionError: [the topology does not contain broker 0] Expected partition <1> to have a healthy leader, but it only has the following brokers: <[BrokerInfoImpl{nodeId=2, host='zeebe-broker-2', port=26501, version=8.1.0-SNAPSHOT, partitions=[PartitionInfoImpl{partitionId=1, role=FOLLOWER, health=HEALTHY}]}, BrokerInfoImpl{nodeId=1, host='zeebe-broker-1', port=26501, version=8.1.0-SNAPSHOT, partitions=[PartitionInfoImpl{partitionId=1, role=FOLLOWER, health=HEALTHY}]}]>
	at io.camunda.zeebe.test.RollingUpdateTest.assertTopologyDoesNotContainerBroker(RollingUpdateTest.java:309)
	at io.camunda.zeebe.test.RollingUpdateTest.lambda$shouldPerformRollingUpdate$10(RollingUpdateTest.java:204)
	at org.awaitility.core.AssertionCondition.lambda$new$0(AssertionCondition.java:53)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:222)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:209)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
</pre>
</details>



**Logs**


<details><summary>Logs</summary>
<pre>
org.awaitility.core.ConditionTimeoutException: Condition with alias 'broker is removed from topology' didn't complete within 10 seconds because assertion condition defined as a io.camunda.zeebe.test.RollingUpdateTest [the topology does not contain broker 0] Expected partition <1> to have a healthy leader, but it only has the following brokers: <[BrokerInfoImpl{nodeId=2, host='zeebe-broker-2', port=26501, version=8.1.0-SNAPSHOT, partitions=[PartitionInfoImpl{partitionId=1, role=FOLLOWER, health=HEALTHY}]}, BrokerInfoImpl{nodeId=1, host='zeebe-broker-1', port=26501, version=8.1.0-SNAPSHOT, partitions=[PartitionInfoImpl{partitionId=1, role=FOLLOWER, health=HEALTHY}]}]>.
	at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:165)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:119)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:31)
	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:895)
	at org.awaitility.core.ConditionFactory.untilAsserted(ConditionFactory.java:679)
	at io.camunda.zeebe.test.RollingUpdateTest.shouldPerformRollingUpdate(RollingUpdateTest.java:204)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:725)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:214)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:210)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:135)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:66)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.executeNonConcurrentTasks(ForkJoinPoolHierarchicalTestExecutorService.java:155)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:129)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.lang.AssertionError: [the topology does not contain broker 0] Expected partition <1> to have a healthy leader, but it only has the following brokers: <[BrokerInfoImpl{nodeId=2, host='zeebe-broker-2', port=26501, version=8.1.0-SNAPSHOT, partitions=[PartitionInfoImpl{partitionId=1, role=FOLLOWER, health=HEALTHY}]}, BrokerInfoImpl{nodeId=1, host='zeebe-broker-1', port=26501, version=8.1.0-SNAPSHOT, partitions=[PartitionInfoImpl{partitionId=1, role=FOLLOWER, health=HEALTHY}]}]>
	at io.camunda.zeebe.test.RollingUpdateTest.assertTopologyDoesNotContainerBroker(RollingUpdateTest.java:309)
	at io.camunda.zeebe.test.RollingUpdateTest.lambda$shouldPerformRollingUpdate$10(RollingUpdateTest.java:204)
	at org.awaitility.core.AssertionCondition.lambda$new$0(AssertionCondition.java:53)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:222)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:209)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
</pre>
</details>

--- a/qa/update-tests/src/test/java/io/camunda/zeebe/test/RollingUpdateTest.java
+++ b/qa/update-tests/src/test/java/io/camunda/zeebe/test/RollingUpdateTest.java
@@ -46,8 +46,8 @@ import org.testcontainers.utility.DockerImageName;
  * These tests are here to detect when rolling update between one version to the next are not
  * possible. If we decide that they aren't between versions, we can disable or adjust assumptions.
  *
- * <p>The important part is that we should be aware whether or not rolling update is possible
- * between versions.
+ * <p>The important part is that we should be aware whether rolling update is possible between
+ * versions.
  */
 final class RollingUpdateTest {
 
@@ -104,7 +104,7 @@ final class RollingUpdateTest {
       broker.start();
 
       Awaitility.await()
-          .atMost(Duration.ofSeconds(5))
+          .atMost(Duration.ofSeconds(120))
           .pollInterval(Duration.ofMillis(100))
           .untilAsserted(() -> assertTopologyContainsUpdatedBroker(client, index));
     }
@@ -121,7 +121,7 @@ final class RollingUpdateTest {
 
       // potentially retry in case we're faster than the deployment distribution
       Awaitility.await("process instance creation")
-          .atMost(Duration.ofSeconds(5))
+          .atMost(Duration.ofSeconds(30))
           .pollInterval(Duration.ofMillis(100))
           .ignoreExceptions()
           .until(() -> createProcessInstance(client), Objects::nonNull)
@@ -134,13 +134,13 @@ final class RollingUpdateTest {
 
     try (final var client = cluster.newClientBuilder().build()) {
       Awaitility.await("broker is removed from topology")
-          .atMost(Duration.ofSeconds(20))
+          .atMost(Duration.ofSeconds(120))
           .pollInterval(Duration.ofMillis(100))
           .untilAsserted(() -> assertTopologyDoesNotContainerBroker(client, brokerId));
 
       for (int i = 0; i < 100; i++) {
         Awaitility.await("process instance creation")
-            .atMost(Duration.ofSeconds(5))
+            .atMost(Duration.ofSeconds(30))
             .pollInterval(Duration.ofMillis(100))
             .ignoreExceptions()
             .until(() -> createProcessInstance(client), Objects::nonNull)
@@ -163,7 +163,7 @@ final class RollingUpdateTest {
       updateBroker(broker);
       broker.start();
       Awaitility.await("updated broker is added to topology")
-          .atMost(Duration.ofSeconds(10))
+          .atMost(Duration.ofSeconds(120))
           .pollInterval(Duration.ofMillis(100))
           .untilAsserted(() -> assertTopologyContainsUpdatedBroker(client, brokerId));
     }
@@ -185,7 +185,7 @@ final class RollingUpdateTest {
       // potentially retry in case we're faster than the deployment distribution
       firstProcessInstanceKey =
           Awaitility.await("process instance creation")
-              .atMost(Duration.ofSeconds(5))
+              .atMost(Duration.ofSeconds(30))
               .pollInterval(Duration.ofMillis(100))
               .ignoreExceptions()
               .until(() -> createProcessInstance(client), Objects::nonNull)
@@ -199,14 +199,14 @@ final class RollingUpdateTest {
         broker.stop();
 
         Awaitility.await("broker is removed from topology")
-            .atMost(Duration.ofSeconds(10))
+            .atMost(Duration.ofSeconds(120))
             .pollInterval(Duration.ofMillis(100))
             .untilAsserted(() -> assertTopologyDoesNotContainerBroker(client, brokerId));
 
         updateBroker(broker);
         broker.start();
         Awaitility.await("updated broker is added to topology")
-            .atMost(Duration.ofSeconds(10))
+            .atMost(Duration.ofSeconds(120))
             .pollInterval(Duration.ofMillis(100))
             .untilAsserted(() -> assertTopologyContainsUpdatedBroker(client, brokerId));
 

